client/state/selectors/has-site-comments.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * External dependencies
+ */
+import { get, find } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+
+/**
+ * Returns true if we have site comments
+ *
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  siteId      The ID of the site we're querying
+ */
+export const hasSiteComments = createSelector(
+	( state, siteId ) => {
+		const comments = get( state, 'comments.items', {} );
+		return !! find( Object.keys( comments ), key => parseInt( key.split( '-', 1 ), 10 ) === siteId );
```

Comment:
the idiom you may be looking for here is `some()`

`some()` return a boolean indicating if at least one item in the collection matches the predicate